File: tools/SharedJavaLibrary/src/network/message/torobot/SetHeadStiffnessMessage.java
package network.message.torobot;

import network.message.RemoteMessage;
import network.message.RemoteMessageToRobot;

public class SetHeadStiffnessMessage extends RemoteMessageToRobot {
  
  public SetHeadStiffnessMessage(float stiffness) {
    super(RemoteMessageToRobot.MessageType.SET_HEAD_STIFFNESS, createData(stiffness));
  }

  private static byte [] createData(float stiffness) {
    // We need 4 bytes for the float
    byte [] bytes = new byte[4];
    RemoteMessage.writeFloat(bytes, 0, stiffness);
    
    return bytes;
  }
}

File: robot/Comm/Messages/Remote/ToRobot/SetHeadStiffnessMessage.cpp
#include "SetHeadStiffnessMessage.h"

#include <cstring>

SetHeadStiffnessMessage::SetHeadStiffnessMessage(float _stiffness)
  : RemoteMessageToRobot(RemoteMessageToRobot::setHeadStiffness, 0, NULL),
    stiffness(_stiffness) {
}

SetHeadStiffnessMessage::~SetHeadStiffnessMessage() {
}

float SetHeadStiffnessMessage::getStiffness() const {
  return stiffness;
}

SetHeadStiffnessMessage const * SetHeadStiffnessMessage::create(unsigned int size, void const *data) {
  // We need exactly 4 bytes
  if (size != 4) {
    return NULL;
  }

  float stiffness = readFloat(reinterpret_cast<char const *>(data));
  return new SetHeadStiffnessMessage(stiffness);
}
